src/components/calendar/calendar.jsx

```diff
@@ -19,55 +20,134 @@ function getHeadCell(iterator) {
   const weekday = iterator.toLocaleDateString(undefined, {
     weekday: 'narrow',
   });
+  const uniqueWeekday = iterator.toLocaleDateString(undefined, {
+    weekday: 'short',
+  });
 
   iterator.setDate(date + 1);
 
   return (
-    <th className={styles.weekday}>
+    <th key={`header-${uniqueWeekday}`} className={styles.weekday}>
       {weekday}
     </th>
   );
 }
 
-function getCellClassName(iterator, month, highlightedDate) {
-  const isHighlightedDate =
-    highlightedDate.getFullYear() === iterator.getFullYear()
-    && highlightedDate.getMonth() === iterator.getMonth()
-    && highlightedDate.getDate() === iterator.getDate();
+function isSameDate(date1, date2) {
+  return (
+    date1.getFullYear() === date2.getFullYear()
+    && date1.getMonth() === date2.getMonth()
+    && date1.getDate() === date2.getDate()
+  );
+}
 
-  if (isHighlightedDate) return styles['highlighted-date'];
+function getCellClassName(iterator, month, highlightedDate) {
+  if (isSameDate(iterator, highlightedDate)) return styles['highlighted-date'];
 
   if (iterator.getMonth() === month) return styles.date;
 
   return styles['not-current-date'];
 }
 
-function getCell(iterator, month, highlightedDate) {
-  const date = iterator.getDate();
-  const className = getCellClassName(iterator, month, highlightedDate);
-
-  iterator.setDate(date + 1);
+function isEndDate(date) {
+  const nextDate = new Date(date.getYear(), date.getMonth(), date.getDate() + 1);
+  return date.getMonth() !== nextDate.getMonth();
+}
 
-  return (
-    <td className={className}>
-      {date}
-    </td>
-  );
+function getDaysInMonth(date, nextMonth) {
+  let month = date.getMonth() + nextMonth;
+  if (isEndDate(date)) month += 1;
+  return new Date(date.getYear(), month, 0).getDate();
 }
 
 function Calendar(props) {
-  const highlightedDate = new Date(props.value ? `${props.value}T00:00` : Date.now());
+  const [selectedDate, setSelectedDate] = useState(props.value ? new Date(`${props.value}T00:00`) : undefined);
+  const highlightedDate = new Date(selectedDate ? selectedDate.getTime() : Date.now());
   const [year, setYear] = useState(highlightedDate.getFullYear());
   const [month, setMonth] = useState(highlightedDate.getMonth());
+  const [active, setActive] = useState(false);
+  const [focusedDate, setFocusedDate] = useState(new Date(props.value ? `${props.value}T00:00` : Date.now()));
```

Comment:
Using the above suggestion

```suggestion
const [focusedDate, setFocusedDate] = useState(date ? date : Date.now());
```